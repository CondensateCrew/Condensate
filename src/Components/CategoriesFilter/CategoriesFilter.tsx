import React from 'react';

const CategoriesFilter:React.FC = () => {
  return (
    <section>
      <button>Genius</button>
      <button>Tech</button>
      <button>Food</button>
      <button>Education</button>
      <button>V</button>
    </section>
  )
}