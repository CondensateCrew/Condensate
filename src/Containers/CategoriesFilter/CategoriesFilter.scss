@import "../../css/variables.scss";
@import "../../css/index.scss";

.filter {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  margin-left: $medium-space;
  width: 35rem; height: 4.25rem;
  position: relative;
}

.closed-filter {
  align-items: center;
  button {
    background: $primary-gradient;
    box-shadow: $external-shadow;
    border-radius: $one-line-border-radius;
    padding: $one-line-padding-space;
    margin-left: $tiny-space;
    height: 4.25rem;
    font-weight: 700;
    &.close-btn {
      img {
        opacity: 0.4;
      }
      &:hover img { opacity: 1; }
    }
    &.chosen-filter {
      background: $success-gradient;
      &:hover {
        background: $success-gradient;
        box-shadow: $external-shadow;
      }
    }
    &#genius {
      background: $genius-gradient;
      margin-left: 0;
      &:hover {
        background: $genius-gradient-mirror;
        box-shadow: $external-shadow, $regular-focus-shadow;
      }
    }
    &:last-child {
      display: -webkit-flex;
      display: -ms-flex;
      display: flex;
      align-items: center;
      justify-content: center;
      width: 4.25rem;
      img {
        width: $big-text;
      }
    }
    &:hover {
      background: $hover-btn-gradient;
      box-shadow: $external-shadow, $regular-focus-shadow;
    }
  }
}

.opened-filter {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $big-blocks-border-radius;
  padding: $one-line-padding-space;
  min-height: 4.25rem;
  position: absolute;
  top: 0; left: 0;
  z-index: 20;
  img {
    transform: rotate(180deg);
    width: $big-text;
    opacity: 0.4;
    &:hover {
      opacity: 1;
    }
  }
  div {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    button {
      @extend %regular-btn;
      font-weight: 400;
      margin: 0;
      margin-right: $small-space;
      margin-bottom: $small-space;
      &.genius-btn {
        background: $genius-gradient;
      }
    }
  }
}
