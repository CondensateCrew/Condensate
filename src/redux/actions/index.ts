import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
