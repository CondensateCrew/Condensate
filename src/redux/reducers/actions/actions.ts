import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
