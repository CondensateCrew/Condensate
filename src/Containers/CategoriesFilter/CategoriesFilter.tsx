import './CategoriesFilter.scss';
import React, { MouseEvent, useState } from 'react';
import { Category, AppStore } from 'interfaces';
import { useDispatch, useSelector } from 'react-redux';
import { addFilter, removeFilter } from 'redux/actions';
import dropdown from 'assets/down.svg';
import close from 'assets/close.svg';

interface Props {
  categories: Category[];
};

const CategoriesFilter:React.FC<Props> = ({categories}) => {
  const [ isClicked, setIsClicked ] = useState<boolean>(false);
  const dispatch = useDispatch();
  const filter = useSelector((state: AppStore) => state.filter);
  const filterCard = (event: MouseEvent<HTMLButtonElement>): void => {
    const { id } = event.target as HTMLButtonElement;
    dispatch(addFilter(id));
    setIsClicked(false);
  };
  const clearFilter = (): void => {
    dispatch(removeFilter());
    setIsClicked(false);
  };
  const allBlocks = categories.map((ctg: Category) => (
    <button key={ctg.id} id={ctg.name} onClick={filterCard}>{ctg.name}</button>
  ));

  const firstCategories = categories.map((ctg: Category, ind: number) => (
    ind < 2 && <button key={ctg.id} id={ctg.name} onClick={filterCard}>{ctg.name}</button>
  ));

  const toggleBlock = (event: React.MouseEvent) => {
    event.preventDefault();
    setIsClicked(!isClicked);
  }

  const closedBlock = (
    <section className="filter closed-filter">
      {
        filter !== ''
        ? <>
          <button className="chosen-filter">{filter}</button>
          <button className="close-btn" onClick={clearFilter}>
            <img src={close} alt="dropdown"/>
          </button>
        </>
        : <>
          <button id="genius" onClick={filterCard}>Genius</button>
          { firstCategories }
          <button className="openning-btn" onClick={toggleBlock}>
            <img src={dropdown} alt="dropdown"/>
          </button>
        </>
      }
    </section>
  );

  const openedBlock = (
    <section className="filter">
      <div className="opened-filter">
        <div>
          <button id="genius" className="genius-btn">Genius</button>
          { allBlocks }
        </div>
        <img src={dropdown} alt="dropdown" onClick={toggleBlock}/>
      </div>
    </section>
  );

  return !isClicked
    ? closedBlock
    : openedBlock;
}

export default CategoriesFilter;
