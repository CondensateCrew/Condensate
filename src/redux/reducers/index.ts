import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
