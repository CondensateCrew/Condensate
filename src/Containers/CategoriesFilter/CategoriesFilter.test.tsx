import React from 'react';
import CategoriesFilter from './CategoriesFilter';
import { shallow } from 'enzyme';
import { addFilter } from 'redux/actions';

jest.mock('redux/actions')

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: () => '',
  useDispatch: () => mockDispatch
}));

describe('CategoriesFilter component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<CategoriesFilter categories={[{id: 1, name: 'Tech'}]} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if btn is clicked', () => {
    wrapper.find('.openning-btn').simulate('click', {preventDefault: jest.fn()});
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addFilter prop when button is clicked', () => {
    wrapper.find('#Tech').simulate('click', {preventDefault: jest.fn(), target: {id: 'Tech'}});
    expect(addFilter).toHaveBeenCalledWith('Tech');
  });
});
