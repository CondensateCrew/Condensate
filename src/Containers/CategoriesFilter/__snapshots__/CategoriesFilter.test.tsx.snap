// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesFilter component should match the snapshot 1`] = `
<section
  className="filter closed-filter"
>
  <button
    id="genius"
    onClick={[Function]}
  >
    Genius
  </button>
  <button
    id="Tech"
    key="1"
    onClick={[Function]}
  >
    Tech
  </button>
  <button
    className="openning-btn"
    onClick={[Function]}
  >
    <img
      alt="dropdown"
      src="down.svg"
    />
  </button>
</section>
`;

exports[`CategoriesFilter component should match the snapshot if btn is clicked 1`] = `
<section
  className="filter"
>
  <div
    className="opened-filter"
  >
    <div>
      <button
        className="genius-btn"
        id="genius"
      >
        Genius
      </button>
      <button
        id="Tech"
        key="1"
        onClick={[Function]}
      >
        Tech
      </button>
    </div>
    <img
      alt="dropdown"
      onClick={[Function]}
      src="down.svg"
    />
  </div>
</section>
`;
